Mathlib/NumberTheory/ModularForms/EisensteinSeries/Finset_Decomposition.lean

```diff
@@ -0,0 +1,108 @@
+/-
+Copyright (c) 2024 Chris Birkbeck. All rights reserved.
```

Comment:
@YaelDillies (sorry for pinging you twice) can you have a look at this file? It can surely be generalized to something crazy instead of `ℤ`, and it is very possible we already have something like this.

Reply:
I will need something like this to reason about percolation. It's unclear how to generalise since what I will need will be in `ι → ℤ`, not `ℤ × ℤ`

Reply:
Actually I found a common generalisation in terms of locally finite ordered rings. I just pushed it. I will also open a separate PR.

Reply:
#10506

Reply:
Oh awesome, thank you!